citar.el

```diff
@@ -901,17 +902,18 @@ was found to have a value, and VALUE is its value."
                   (cons field value)))
               fields)))
 
-(defun citar-get-display-value (fields citekey-or-entry)
+(defun citar-get-display-value (fields citekey-or-entry &optional template)
   "Return the first non nil value for CITEKEY-OR-ENTRY among FIELDS .
 
-The value is transformed using `citar-display-transform-functions'"
+The value is transformed using `citar-display-transform-functions' for TEMPLATE."
   (let ((fieldvalue (citar-get-field-with-value fields citekey-or-entry)))
     (seq-reduce (lambda (string fun)
+                  ;; FIX: can't get this to work, and don't understand this line.
```

Comment:
What is the problem here? `fun` is badly named since it will actually be an element of `citar-display-transform-functions` but I probably couldn't (still can't) think of a name.

Reply:
I need to remove the comment, as it does not work.

But I still don't understand the line immediately below it.

Reply:
> I need to remove the comment, as it does not work.

I assume the not is a typo?

> But I still don't understand the line immediately below it.

`fun` is an element of `citar-display-transform-functions` so `(car fun)` is either a list of field names or it is `t` where the `t` means the function which is in the `(cdr fun)` should be applied to all fields.

Reply:
Yeah; typo, now fixed.

Thanks for explanation!